attestation/yubiattest/parse.go

```diff
@@ -668,20 +666,40 @@ func parsePublicKey(algo x509.PublicKeyAlgorithm, keyData *publicKeyInfo) (inter
 		if len(rest) != 0 {
 			return nil, errors.New("x509: trailing data after ECDSA parameters")
 		}
-		namedCurve := namedCurveFromOID(*namedCurveOID)
-		if namedCurve == nil {
-			return nil, errors.New("x509: unsupported elliptic curve")
+		c, err := ecdhCurveFromOID(*namedCurveOID)
+		if err != nil {
+			return nil, err
 		}
-		x, y := elliptic.Unmarshal(namedCurve, asn1Data)
-		if x == nil {
-			return nil, errors.New("x509: failed to unmarshal elliptic curve point")
+		p, err := c.NewPublicKey(asn1Data)
+		if err != nil {
+			return nil, errors.New("x509: parsing public key for ECDSA")
 		}
-		pub := &ecdsa.PublicKey{
-			Curve: namedCurve,
-			X:     x,
-			Y:     y,
+		encodedPubKey := p.Bytes()
+		// The elliptic.Unmarshal() function has been deprecated in Go.
+		// As a workaround, we manually parse the public key by its index.
+		// For more details, see: https://github.com/golang/go/issues/63963
+		switch p.Curve() {
+		case ecdh.P256():
+			return &ecdsa.PublicKey{
+				Curve: elliptic.P256(),
+				X:     big.NewInt(0).SetBytes(encodedPubKey[1:33]),
```

Comment:
can we add a comment around not needing length check as `Bytes()` ensures the size is 133 https://github.com/golang/go/blob/release-branch.go1.23/src/crypto/ecdh/ecdh.go#L75-L80 ?